client/cli_main.py

```diff
@@ -228,11 +268,91 @@ def main():
+                            reading = True
+                            if len(thoughts) == 0:
+                                print("No thoughts found for this user.")
+                            else:
+                                while reading:
+                                    print("Please choose a thought to read by entering its number or type B to go back:\n")
+                                    i = 0
+                                    for thought in thoughts:
+                                        print(f"{i + 1}. TITLE: {thought['title']}\t RATING: {thought['rating']}")
+                                        i += 1
+                                    try:
+                                        thought_choice = input("\nEnter thought number: ")
+                                        if thought_choice == "b" or thought_choice == "B":
+                                            reading = False
+                                        else:
+                                            thought_num = int(thought_choice)
+                                            selected_thought = thoughts[thought_num - 1]
+                                            print(f"\nTITLE:  {selected_thought['title']}")
+                                            print(f"RATING: {selected_thought['rating']}\n")
+                                            decrypted_message = decrypt_message(selected_thought['content'], encrypted_sym_key)
+                                            print(f"MESSAGE:  {decrypted_message}\n\n")
+                                            key_path = os.path.join(os.path.dirname(__file__), 'keys', 'message.key')
+                                            with open(key_path, 'rb') as key_file:
+                                                key = str(key_file.read())
+                                            update_rating_for_thought(server_url, key)
+                                            finished_reading = input("B: Finished reading ")
+                                            if finished_reading == "b" or finished_reading == "B":
+                                                reading = False
+                                    except FileNotFoundError as err:
+                                        print("Error decrypting message, you may need to generate your keys still!\nError:", err)
+                                    except ValueError as err:
+                                        print("Please restart the programme to register your keys!\nError:", err)
+                                    except IndexError as err:
+                                        print("You selected a non-existant thought number.\nError:", err)
+                    elif sub_choice == "8":
+                        friend_username = input("Please enter your friends username: \n")
+                        get_user_conversation(server_url, friend_username)
```

Comment:
I'm getting this error (whole output, error near bottom (404 doubled):
```txt
@Lukas-Batema ➜ /workspaces/PeerBrain (development) $ python client/cli_main.py

Welcome to version 0.2 of the Brainwaves P2P client!
-------------------------------------------------------------------
The currently set server url is https://peerbrain.teckhawk.be/
-------------------------------------------------------------------

Please select what you want to do from the menu below.
1. Log in to the current server
2. Register account on the current server
3. Change server
4. Reset your password
Q to exit
>> 1

Logged in with valid token


MAIN MENU:


Please choose an option:

1. TECHNICAL ACTIONS
2. ACCOUNT ACTIONS
3. LOG OUT
Q to exit
>> 2

ACCOUNT MENU:


Please choose an option:

1. Check your account details
--------------------------------
2. Create a message
3. Show all messages from a friend
--------------------------------
4. Add a friend
5. Check friends list
6. Remove a friend from your friend list
--------------------------------
7. New thought display function test
8. Private conversations(DM)
B to return to main menu
>> 8
Please enter your friends username: 
admin



--------
{'detail': 'Not Found'}
--------
{'detail': 'Not Found'}
Error: 404

1. Yes
2. No
Do you want to reply to admin? 1
Enter your message: joe
Something went wrong with sending your message!


ACCOUNT MENU:


Please choose an option:

1. Check your account details
--------------------------------
2. Create a message
3. Show all messages from a friend
--------------------------------
4. Add a friend
5. Check friends list
6. Remove a friend from your friend list
--------------------------------
7. New thought display function test
8. Private conversations(DM)
B to return to main menu
>> b
Returning to main menu...

MAIN MENU:


Please choose an option:

1. TECHNICAL ACTIONS
2. ACCOUNT ACTIONS
3. LOG OUT
Q to exit
>> q
Goodbye!
```
Will that be fixed with the dev database that I don't have access to?

Reply:
Okay, it should just need these changes (basically renaming the option 7 of main option 2)

Reply:
done :-)